src/Components/Endpoints/src/Rendering/EndpointHtmlRenderer.EventDispatch.cs

```diff
@@ -78,9 +78,18 @@ private void SetNotFoundResponse(object? sender, EventArgs args)
     {
         if (_httpContext.Response.HasStarted)
         {
-            throw new InvalidOperationException("Cannot set a NotFound response after the response has already started.");
+            // We cannot set a NotFound code after the response has already started
+            var navigationManager = _httpContext.RequestServices.GetRequiredService<NavigationManager>();
+            if (navigationManager is null)
+            {
+                throw new InvalidOperationException("The NavigationManager service is not available. Cannot navigate to a 404 page.");
+            }
+            var notFoundUri = $"{navigationManager.BaseUri}not-found";
+            navigationManager.NavigateTo(notFoundUri);
```

Comment:
Hmm, is this bit the part that handles streaming rendering?

Reply:
Yes. Streaming rendering in case the streaming started. Why?

Reply:
I don't see a test for it.

I would expect this to be something like a component with `[StreamingRendering]` applied to it and something like

```razor
<p>Some content</p>
@code {
protected async Task OnInitializedAsync() {
  await Task.Yield();
  nav.NotFound();
}
}
```

Reply:
Thanks, I missed streaming tests. That should be done. Do you think we should have also the tests we do right now copied-pasted to app that uses `RazorComponentEndpointsNoInteractivityStartup`? Right now, we are testing with global interactivity only.